tequilapi/endpoints/service.go

```diff
@@ -0,0 +1,230 @@
+// description: ServiceList provides a list of running services on the node.
+// responses:
+//   200:
+//     description: List of running services
+//     schema:
+//       "$ref": "#/definitions/ServiceListDTO"
+func (se *ServiceEndpoint) ServiceList(resp http.ResponseWriter, _ *http.Request, _ httprouter.Params) {
+	statusResponse := toServiceListResponse(se.serviceState)
+	utils.WriteAsJSON(statusResponse, resp)
+}
+
+// ServiceGet provides info for requested service on the node.
+// swagger:operation GET /services/:id Service serviceGet
+// ---
+// summary: Information about service
+// description: ServiceGet provides info for requested service on the node.
+// responses:
+//   200:
+//     description: Service detailed information
+//     schema:
+//       "$ref": "#/definitions/ServiceInfoDTO"
+//   404:
+//     description: Service not found
+//     schema:
+//       "$ref": "#/definitions/ErrorMessageDTO"
+func (se *ServiceEndpoint) ServiceGet(resp http.ResponseWriter, _ *http.Request, params httprouter.Params) {
+	statusResponse := toServiceInfoResponse(se.serviceState)
+	utils.WriteAsJSON(statusResponse, resp)
+}
+
+// ServiceStart starts requested service on the node.
+// swagger:operation POST /service Service serviceStart
+// ---
+// summary: Starts service
+// description: Provider starts serving new service to consumers
+// parameters:
+//   - in: body
+//     name: body
+//     description: Parameters in body (providerID) required for starting new service
+//     schema:
+//       $ref: "#/definitions/ServiceRequestDTO"
+// responses:
+//   201:
+//     description: Service started
+//     schema:
+//       "$ref": "#/definitions/ServiceInfoDTO"
+//   400:
+//     description: Bad request
+//     schema:
+//       "$ref": "#/definitions/ErrorMessageDTO"
+//   409:
+//     description: Conflict. Service is already running
+//     schema:
+//       "$ref": "#/definitions/ErrorMessageDTO"
+//   422:
+//     description: Parameters validation error
+//     schema:
+//       "$ref": "#/definitions/ValidationErrorDTO"
+//   499:
+//     description: Service was cancelled
+//     schema:
+//       "$ref": "#/definitions/ErrorMessageDTO"
+//   500:
+//     description: Internal server error
+//     schema:
+//       "$ref": "#/definitions/ErrorMessageDTO"
+func (se *ServiceEndpoint) ServiceStart(resp http.ResponseWriter, req *http.Request, params httprouter.Params) {
+	sr, err := toServiceRequest(req)
+	if err != nil {
+		utils.SendError(resp, err, http.StatusBadRequest)
+		return
+	}
+
+	errorMap := validateServiceRequest(sr)
+	if errorMap.HasErrors() {
+		utils.SendValidationErrorMessage(resp, errorMap)
+		return
+	}
+
+	se.serviceState = service.Running
+
+	resp.WriteHeader(http.StatusCreated)
+	se.ServiceGet(resp, req, params)
+}
+
+// ServiceStop stops service on the node.
+// swagger:operation DELETE /service/:id Service serviceStop
+// ---
+// summary: Stops service
+// description: Stops current service
```

Comment:
So if I understand correctly, if service is `Starting`, than this will try to cancel service before it starts?
If so, can we add a notice for this case here?

Reply:
I think it should stop both `Running` and `Starting` states of the service.

Let's say you were using the system as a consumer only, then decided to provide `openvpn` service. You need to start `openvpn` service by `POST "openvpn" to /services`.

After some time you decided to stop providing `openvpn` and switch to `wireguard`. You need to do `DELETE /services/openvpn` and `POST "wireguard" to /services`. 

All these steps should be done without node restart.